### PR TITLE
Plane:prevent in landing prearm fails if booting without rc

### PR DESCRIPTION
**Pull request by Hwurzburg**
addresses #23180
SITL tested....only issue is that if the mission starts with a DO_LAND_START resetting the mission on the first valid RC does not solve the issue.....an acceptable limitation in my opinion

### CODE REVIEW

**Comment by timtuxworth** [comment]
Fixes this too am I right? https://discuss.ardupilot.org/t/advance-mission-planning-issue/88116/17?u=timtuxworth

**Comment by IamPete1** [comment]
I think this should be handled on the mode change. I'm quite weary of unintended consequences here tho. Don't want to reset someone's mission after a watchdog for example. 

Copter gets round this by just not taking any rc fail-safe actions while disarmed, copter is not going to be flying when disarmed tho plane could be.

File: ArduPlane/Plane.h [hunk 1]
```diff
@@ -1041,6 +1041,7 @@ class Plane : public AP_Vehicle {
     void trim_radio();
     bool rc_throttle_value_ok(void) const;
     bool rc_failsafe_active(void) const;
+    bool seen_valid_rc = false;
```
Comment (peterbarker) [inline ArduPlane/Plane.h:1044]: Don't need to zero this.  Costs flash.

Comment (Hwurzburg) [reply ArduPlane/Plane.h]: fixed

**Comment by Hwurzburg** [comment]
> I think this should be handled on the mode change. I'm quite weary of unintended consequences here too. Don't want to reset someone's mission after a watchdog for example.
do watchdog's reset variables? or disarm? the mission reset here would require both of those two things to happen

**Comment by IamPete1** [comment]
> do watchdog's reset variables? or disarm? the mission reset here would require both of those two things to happen

Yes, it does both. 

I think there is already a has had valid flag/timestamp down in RC Channels.

**Comment by Hwurzburg** [comment]
okay so a watchdog resets the FC....once that is done, you still have RC going, and you have to manually re-arm, correct? so this is a normal reboot with RC present....so no mission reset....the  landing flag would have been cleared by the reboot...no issue I believe

**Comment by Hwurzburg** [comment]
as it stands this does not fix the issue of autolanding with disarm..

**Comment by Hwurzburg** [comment]
after chatting with Pete, he is posting a better PR and I am closing this....I will work on the issue of disarming after landing and getting rid of that prearm #23189

**Pull request closed without merging**